frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -605,13 +621,52 @@ private ContainerLaunchContext setupContainerLaunchContext(TaskStatus taskStatus
     return launchContext;
   }
 
+  // This function is to convert task's containerPorts from List<Range> format to string format
+  // The string format is "httpPort:80,81,82;sshPort:1021,1022,1023;"
+
+  private String setupPortsEnvironment(TaskStatus taskStatus) {
```

Comment:
>setupPortsEnvironment [](start = 17, length = 21)

Can we make it a util to convert from List<ValueRange> to portString?

Reply:
The function has portDefinition, which is a map.  it is not a simple List<ValueRange> to string, it need do  port label "Join"

---
In reply to: [173104703](https://github.com/Microsoft/pai/pull/318#discussion_r173104703) [](ancestors = 173104703)